### DISCUSSION TITLE
Local Key invalid

### DISCUSSION THREAD

**Discussion by Rem9876543** [iRayanKhan/homebridge-tuya#413] [New device help]
Hi, 
I get the following Local Key via the Tuya IOT platform for a simple dimmer:
    _"local_key": "Y6+7>{IHKhY'`f^Q",_

It's not an hexadecimal number like the examples I saw so far, and the homebridge-tuya plugin does not like it:
_[6/14/2023, 9:57:19 AM] [TuyaLan] Y6+7>{IHKhY'****, key for Office Dimmer (bf8adb8f27d201c2abiapm), is not a valid key._

Is the key wrong or is the homebridge-tuya plugin wrong for expecting an hexadecimal number?
If the key is wrong, why would the Tuya IOT platform return that Local Key?

Thanks heaps

**Comment by AndShezar** [comment]
The same thing happens to me, with error "is not a valid key"

any information on this problem?

**Comment by serwistech** [comment]
Hi, same for me. 
Is there a chance to solve the problem?

**Reply by serwistech** [reply]
I disabled key validation in index.js and it finds the device. Unfortunately, there is another problem

`[Tuya] Socket had a problem and will reconnect to SwitchKuchnia (Error: ERR_PING_TIMED_OUT)`

Even after adding IP to config.

**Comment by remussamoila** [comment]
Hi!
Check this solution if you are comfortable editing index.js file
[https://github.com/iRayanKhan/homebridge-tuya/issues/409#issuecomment-1583165740](url)

**Reply by serwistech** [reply]
I added, devices are visible but cannot be controlled. They also don't report the status.

**Comment by 05TEVE** [comment]
Hey guys, I have pushed a beta version for testing that should resolve this issue. Can you please test the 3.1.0-beta.2 version and let me know if it resolves the issue?

**Comment by remussamoila** [comment]
yes, 3.1 version can use the new format pf tuya keys

**Comment by Aljeksandr** [comment]
Local key all time change after re/connecting link disconnect app it's so bad
